### ISSUE TITLE
Changing user base path breaks ability to preview

### ISSUE THREAD

**Issue by EzraBlint** [alist-org/alist#1636]
### Please make sure of the following things

- [X] I have read the [documentation](https://alist-doc.nn.ci).
- [X] I'm sure there are no duplicate issues or discussions.
- [X] I'm sure it's due to `alist` and not something else(such as `Dependencies` or `Operational`).

### Alist Version / Alist 版本

v3.0.0 beta.1

### Driver used / 使用的存储驱动

Local

### Describe the bug / 问题描述

Preview ability stops functioning properly when user account Base Path is changed.

Testing
I have tried this with both a guest account and newly created account to rule out any issue with guest account. 
Setting accounts to have full permissions also did not fix the issue.
When attempting to download file, this does work without issue so issue appears to be specific to the preview feature.

When base path is set to root, everything works without issue leading me to believe there is an issue with preview interpreting file path when base path is changed for user.

Screenshots below show evidence of issue 

With Base Path set as root evidence to show file initially working,
![Text File - with normal basepath](https://user-images.githubusercontent.com/8010518/189511983-4d8a208e-e010-4486-a045-1748001e71c3.jpg)

When Base Path changed to subdirectory (Public)
![Text File - error example](https://user-images.githubusercontent.com/8010518/189511987-bd7ad8ce-8a4b-4868-8ec1-fec6a0822c2c.jpg)
![Audio Test](https://user-images.githubusercontent.com/8010518/189511986-d4dfe2b4-65b8-4ded-9fde-aea8a08748c7.jpg)

Screenshot to show Base path for user when not working.
![Base Path - Change](https://user-images.githubusercontent.com/8010518/189512016-e561d2c1-b013-4b7b-8b5a-eba786ead462.jpg)

also noticed a typo for password when editing user (typed pssword)
![Password typo](https://user-images.githubusercontent.com/8010518/189512623-8a58a625-aa02-45ca-8c4c-f097fcb8faa1.png)

Thank you very much for your time and effort on this cool project. I am also aware this is currently a beta version I just thought I would flag this.

### Reproduction / 复现链接

First go to storages and add a storage location (in this case Driver = "Local"), 
I set Mount Path: /
I set root folder: /mnt/data
all other settings left as default.
in /mnt/data are 2 folders. Private and Public

In users, create a user and set "Base path" to a subdirectory (in this case Public)
log in as new user and attempt to preview a file.

### Logs / 日志

```shell
[GIN] 2022/09/11 - 04:42:20 | 200 |      69.519..s |   REDACTED | GET      "/undefined"                                                                
^[[31mERRO^[[0m[2022-09-11 04:42:20] failed link /Test-Audio.mp3: object not found                                                                          
github.com/alist-org/alist/v3/internal/op.Get                                                                                                               
        /app/internal/op/fs.go:127                                                                                                                           
github.com/alist-org/alist/v3/internal/op.Link                                                                                                              
        /app/internal/op/fs.go:138                                                                                                                          
github.com/alist-org/alist/v3/internal/fs.link                                                                                                              
        /app/internal/fs/link.go:16                                                                                                                         
github.com/alist-org/alist/v3/internal/fs.Link                                                                                                              
        /app/internal/fs/fs.go:35                                                                                                                           
github.com/alist-org/alist/v3/server/handles.Proxy                                                                                                               
        /app/server/handles/down.go:64                                                                                                                      
github.com/alist-org/alist/v3/server/handles.Down                                                                                                           
        /app/server/handles/down.go:27                                                                                                                      
github.com/gin-gonic/gin.(*Context).Next                                                                                                                           
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/context.go:173                                                                                     
github.com/alist-org/alist/v3/server/middlewares.Down                                                                                                                   
        /app/server/middlewares/down.go:38                                                                                                                  
github.com/gin-gonic/gin.(*Context).Next                                                                                                                    
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/context.go:173                                                                                     
github.com/alist-org/alist/v3/server/middlewares.StoragesLoaded                                                                                             
        /app/server/middlewares/check.go:11                                                                                                                        
github.com/gin-gonic/gin.(*Context).Next                                                                                                                    
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/context.go:173                                                                                     
github.com/gin-gonic/gin.CustomRecoveryWithWriter.func1                                                                                                     
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/recovery.go:101                                                                                    
github.com/gin-gonic/gin.(*Context).Next                                                                                                                    
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/context.go:173                                                                                     
github.com/gin-gonic/gin.LoggerWithConfig.func1                                                                                                             
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/logger.go:240                                                                                      
github.com/gin-gonic/gin.(*Context).Next                                                                                                                           
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/context.go:173                                                                                     
github.com/gin-gonic/gin.(*Engine).handleHTTPRequest                                                                                                        
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/gin.go:613                                                                                         
github.com/gin-gonic/gin.(*Engine).ServeHTTP                                                                                                                
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/gin.go:569                                                                                         
net/http.serverHandler.ServeHTTP                                                                                                                            
        /usr/lib/go/src/net/http/server.go:2947                                                                                                             
net/http.(*conn).serve                                                                                                                                      
        /usr/lib/go/src/net/http/server.go:1991                                                                                                             
runtime.goexit                                                                                                                                              
        /usr/lib/go/src/runtime/asm_amd64.s:1594 
failed to get file                                                                                                                                          
[GIN] 2022/09/11 - 04:42:20 | 200 |    3.129126ms |   REDACTED | GET      "/d/Test-Audio.mp3?sign=ukHE3sg1BGegoQ7Zy2T_AAlLwPTKwB6B2jIhjpT1pLs=:0"      
[GIN] 2022/09/11 - 04:42:22 | 302 |      20.173..s |   REDACTED | GET      "/favicon.ico"                                                              
[GIN] 2022/09/11 - 04:42:22 | 200 |    3.409673ms |   REDACTED | POST     "/api/fs/list"                                                               
[GIN] 2022/09/11 - 04:42:23 | 302 |      26.865..s |   REDACTED | GET      "/favicon.ico"                                                              
[GIN] 2022/09/11 - 04:42:23 | 200 |    3.399537ms |   REDACTED | POST     "/api/fs/get"                                                                
^[[31mERRO^[[0m[2022-09-11 04:42:23] failed link /Something Public.txt: object not found                                                                    
github.com/alist-org/alist/v3/internal/op.Get                                                                                                               
        /app/internal/op/fs.go:127                                                                                                                          
github.com/alist-org/alist/v3/internal/op.Link                                                                                                               
        /app/internal/op/fs.go:138                                                                                                                          
github.com/alist-org/alist/v3/internal/fs.link                                                                                                              
        /app/internal/fs/link.go:16                                                                                                                         
github.com/alist-org/alist/v3/internal/fs.Link                                                                                                              
        /app/internal/fs/fs.go:35                                                                                                                           
github.com/alist-org/alist/v3/server/handles.Proxy                                                                                                          
        /app/server/handles/down.go:64                                                                                                                           
github.com/gin-gonic/gin.(*Context).Next                                                                                                                    
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/context.go:173                                                                                     
github.com/alist-org/alist/v3/server/middlewares.Down                                                                                                       
        /app/server/middlewares/down.go:38                                                                                                                         
github.com/gin-gonic/gin.(*Context).Next                                                                                                                    
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/context.go:173                                                                                                 
github.com/alist-org/alist/v3/server/middlewares.StoragesLoaded                                                                                             
        /app/server/middlewares/check.go:11                                                                                                                 
github.com/gin-gonic/gin.(*Context).Next                                                                                                                    
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/context.go:173                                                                                     
github.com/gin-gonic/gin.CustomRecoveryWithWriter.func1                                                                                                            
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/recovery.go:101                                                                                    
github.com/gin-gonic/gin.(*Context).Next                                                                                                                    
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/context.go:173                                                                                     
github.com/gin-gonic/gin.LoggerWithConfig.func1                                                                                                             
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/logger.go:240                                                                                      
github.com/gin-gonic/gin.(*Context).Next                                                                                                                    
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/context.go:173                                                                                     
github.com/gin-gonic/gin.(*Engine).handleHTTPRequest                                                                                                        
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/gin.go:613                                                                                                
github.com/gin-gonic/gin.(*Engine).ServeHTTP                                                                                                                
        /root/go/pkg/mod/github.com/gin-gonic/gin@v1.8.0/gin.go:569                                                                                         
net/http.serverHandler.ServeHTTP                                                                                                                            
        /usr/lib/go/src/net/http/server.go:2947                                                                                                             
net/http.(*conn).serve                                                                                                                                      
        /usr/lib/go/src/net/http/server.go:1991                                                                                                             
runtime.goexit                                                                                                                                              
        /usr/lib/go/src/runtime/asm_amd64.s:1594                                                                                                            
failed to get file                                                                                                                                          
[GIN] 2022/09/11 - 04:42:23 | 200 |    2.756347ms |   REDACTED | GET      "/p/Something Public.txt?sign=fneAUQK1iAN5DBbJCGE7T2tvmas7z9uOra3sQvdd_AA=:0"
[GIN] 2022/09/11 - 04:42:24 | 302 |      20.246..s |   REDACTED | GET      "/favicon.ico"                                                              
[GIN] 2022/09/11 - 04:42:24 | 200 |    2.288924ms |   REDACTED | POST     "/api/fs/list"              
```

**Comment by Xhofe** [comment]
I believe this has been fixed at https://github.com/alist-org/alist-web/commit/a37dc592952af1c693edff86832ea4ad2810acf7, please wait for the release

**Comment by Xhofe** [comment]
Fixed in [v3.0.0-beta.2](https://github.com/alist-org/alist/releases/tag/v3.0.0-beta.2)
